extensions/common/vault/vault-gcp/src/test/java/org/eclipse/edc/vault/gcp/GcpSecretManagerVaultTest.java

```diff
@@ -0,0 +1,277 @@
+/*
+ *  Copyright (c) 2023 Google LLC
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Google LCC - Initial implementation
+ *
+ */
+
+package org.eclipse.edc.vault.gcp;
+
+import com.google.api.gax.rpc.NotFoundException;
+import com.google.api.gax.rpc.StatusCode;
+import com.google.cloud.secretmanager.v1.ProjectName;
+import com.google.cloud.secretmanager.v1.Replication;
+import com.google.cloud.secretmanager.v1.Secret;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretName;
+import com.google.cloud.secretmanager.v1.SecretVersionName;
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.mockito.ArgumentMatchers;
+
+import java.util.ArrayList;
+import java.util.Random;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.TestInstance.Lifecycle;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@TestInstance(Lifecycle.PER_CLASS)
+class GcpSecretManagerVaultTest {
+
+    class TestStatusCode implements StatusCode {
+        @Override
+        public Integer getTransportCode() {
+            return Integer.valueOf(0);
+        }
+
+        @Override
+        public Code getCode() {
+            return Code.OK;
+        }
+    }
+
+    private static final String TEST_REGION = "europe-west3";
+    private static final String TEST_PROJECT = "project";
+    private static final String VALID_KEY = "test";
+    private static final String INVALID_KEY = "test=";
+    private static final String SANITIZED_KEY = "test-_06924DEB";
+
+    private static final int MAX_KEY_LENGTH = 255;
+
+    private final Monitor monitor = mock(Monitor.class);
```

Comment:
no need to pass the class obj to the `mock` method when the type is explicitly defined.